### PR TITLE
Only show stderr of Nix failures when a failure occurred

### LINKED ISSUE #30: Avoid alias warnings
When doing the Nix evaluation of packages in `pkgs/by-name`, the tool runs with an inherited stderr:

https://github.com/NixOS/nixpkgs-check-by-name/blob/65a04d8c47378684ecd4ef41c07fe8d1c34e480a/src/eval.rs#L129-L130

This means that errors get shown when they happen, but it also means that trace get shown even if there's no error. This can be seen e.g. [here](https://github.com/NixOS/nixpkgs/actions/runs/8442701004/job/23124705090):

```
trace: warning: eww now can build for X11 and wayland simultaneously, so `eww-wayland` is deprecated, use the normal `eww` package instead.
trace: warning: eww now can build for X11 and wayland simultaneously, so `eww-wayland` is deprecated, use the normal `eww` package instead.
trace: warning: nvtop has been renamed to nvtopPackages.full
trace: warning: nvtop-amd has been renamed to nvtopPackages.amd
trace: warning: nvtop-intel has been renamed to nvtopPackages.intel
trace: warning: nvtop-msm has been renamed to nvtopPackages.msm
trace: warning: nvtop-nvidia has been renamed to nvtopPackages.nvidia
trace: warning: nvtop has been renamed to nvtopPackages.full
trace: warning: nvtop-amd has been renamed to nvtopPackages.amd
trace: warning: nvtop-intel has been renamed to nvtopPackages.intel
trace: warning: nvtop-msm has been renamed to nvtopPackages.msm
trace: warning: nvtop-nvidia has been renamed to nvtopPackages.nvidia
trace: warning: writeReferencesToFile is deprecated in favour of writeClosure
trace: warning: writeReferencesToFile is deprecated in favour of writeClosure
Validated successfully
```

This only happens for warnings on top-level packages that were defined with a warning in `aliases.nix`, e.g. 

https://github.com/NixOS/nixpkgs/blob/74518d2bf610de186cc000d1d479cb975e5a9ffc/pkgs/top-level/aliases.nix#L331

https://github.com/NixOS/nixpkgs/blob/74518d2bf610de186cc000d1d479cb975e5a9ffc/pkgs/top-level/aliases.nix#L871-L876

https://github.com/NixOS/nixpkgs/blob/ab14a429cb1a6cc7b7b843a8ad2661ad1570e57c/pkgs/build-support/trivial-builders/default.nix#L629-L631

For now I don't think we should act on this, but I'm not convinced this is a good pattern to use in general, it's very new.

### PR DESCRIPTION
**Pull request by infinisil**
This fixes #30

See commit messages and comments for more details.

### CODE REVIEW

File: src/eval.rs [hunk 1]
```diff
@@ -128,14 +130,13 @@ pub fn check_values(
     let mut command = process::Command::new("nix-instantiate");
     command
         // Inherit stderr so that error messages always get shown
```
Comment (willbush) [inline src/eval.rs:132]: This comment is out-of-date

File: src/main.rs [hunk 2]
```diff
@@ -187,7 +183,7 @@ mod tests {
             }
 
             let expected_errors = fs::read_to_string(path.join("expected"))
-                .expect("No expected file for test {name}");
+                .expect(&format!("No expected file for test {name}"));
```
Comment (willbush) [inline src/main.rs:186]: I'm seeing a clippy warning here:

```
warning: use of `expect` followed by a function call
   --> src/main.rs:186:18
    |
186 |                 .expect(&format!("No expected file for test {name}"));
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `unwrap_or_else(|_| panic!("No expected file for test {name}"))`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#expect_fun_call
    = note: `#[warn(clippy::expect_fun_call)]` on by default

warning: `nixpkgs-check-by-name` (bin "nixpkgs-check-by-name" test) generated 1 warning (run `cargo clippy --fix --bin "nixpkgs-check-by-name" --tests` to apply 1 suggestion)
    Finished dev [unoptimized + debuginfo] target(s) in 2.35s
```

**Review by willbush — CHANGES_REQUESTED** [review]

**Comment by infinisil** [comment]
@willbush Thanks, addressed!

**Review by philiptaron — APPROVED** [review]
Works for me.

**Pull request merged**